libraries/AP_Mount/AP_Mount_Siyi.h

```diff
@@ -0,0 +1,192 @@
+/*
+  Siyi gimbal driver using custom serial protocol
+
+  Packet format (courtesy of Siyi's SDK document)
+
+  -------------------------------------------------------------------------------------------
+  Field     Index   Bytes       Description
+  -------------------------------------------------------------------------------------------
+  STX       0       2           0x5566: starting mark
+  CTRL      2       1           bit 0: need_ack.  set if the current data packet needs ack
+                                bit 1: ack_pack.  set if the current data packate IS an ack
+                                bit 2-7: reserved
+  Data_len  3       2           Data field byte length.  Low byte in the front
+  SEQ       5       2           Frame sequence (0 ~ 65535).  Low byte in the front.  May be used to detect packet loss
+  CMD_ID    7       1           Command ID
+  DATA      8       Data_len    Data
+  CRC16             2           CRC16 check the complete data package.  Low byte in the front
+ */
+
+#pragma once
+
+#include "AP_Mount_Backend.h"
+
+#ifndef HAL_MOUNT_SIYI_ENABLED
+#define HAL_MOUNT_SIYI_ENABLED HAL_MOUNT_ENABLED
+#endif
+
+#if HAL_MOUNT_SIYI_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_Common/AP_Common.h>
+
+#define AP_MOUNT_SIYI_PACKETLEN_MAX     22  // maximum number of bytes in a packet sent to or received from the gimbal
+
+class AP_Mount_Siyi : public AP_Mount_Backend
+{
+
+public:
+    // Constructor
+    AP_Mount_Siyi(AP_Mount &frontend, AP_Mount_Params &params, uint8_t instance);
+
+    /* Do not allow copies */
+    AP_Mount_Siyi(const AP_Mount_Siyi &other) = delete;
+    AP_Mount_Siyi &operator=(const AP_Mount_Siyi&) = delete;
```

Comment:
```suggestion
    CLASS_NO_COPY(AP_Mount_Siyi)
```

Reply:
txs..  There are [many other examples of where we don't use this macro so I've created an issue](https://github.com/ArduPilot/ardupilot/issues/21837).